README.md

```diff
@@ -1,77 +1,185 @@
+2. **Sample Sheet (CSV)**
 
-## Optional: Uploading Results to Synapse
-To upload results to Synapse, run the workflow with the following parameters:
+Example `input.csv`:
+```csv
+repo_url,description
+https://github.com/PythonOT/POT.git,Python Optimal Transport Library
+https://github.com/RabadanLab/TARGet.git,TARGet Analysis Tool
+```
+
+### Running the Workflow
 
+#### Basic Analysis
+```bash
+nextflow run main.nf --repo_url https://github.com/example/repo.git
+```
+
+#### With AI Analysis
 ```bash
 nextflow run main.nf \
     --repo_url https://github.com/example/repo.git \
-    --upload_to_synapse true\
-    --synapse_folder_id syn64626421
+    --synapse_agent_id LOWYSX3QSQ
+```
+
+#### With Sample Sheet
+```bash
+nextflow run main.nf --sample_sheet input.csv
 ```
-Ensure your Synapse credentials are properly set up (e.g., by mounting your .synapseConfig file).
 
-## Tools You Can Test With
+> [!NOTE]
+> When using AI Analysis or Synapse integration, ensure you have:
+> - Valid Synapse authentication token
+> - Proper Synapse configuration
+> - Synapse agent ID for AI analysis (e.g., LOWYSX3QSQ)
+> - Correct folder ID with write permissions (for upload)
+
+## Output
+
+The workflow generates several output files in the `results` directory:
+
+- `<repo_name>_ai_analysis.json`: AI-powered qualitative summary and recommendations (final report)
+- `almanack_results.json`: Detailed metrics from Almanack analysis
+- `joss_report_<repo_name>.json`: JOSS criteria evaluation metrics
+- `test_results_<repo_name>.json`: Test execution results and coverage metrics
+
+> [!NOTE]
+> The AI analysis report provides a high-level qualitative summary and actionable recommendations. For detailed metrics and specific measurements, refer to the other output files.
+
+## Development Status
+
+> [!WARNING]
+> The AI Analysis component is currently in beta. Results may vary and the interface is subject to change.
+
+> [!IMPORTANT]
+> Synapse integration requires proper authentication and permissions setup.
+
+## Example Repositories
+
+| Repository | Description | Expected Status |
+|------------|-------------|----------------|
+| [PythonOT/POT](https://github.com/PythonOT/POT) | Python Optimal Transport Library | All checks pass |
+| [RabadanLab/TARGet](https://github.com/RabadanLab/TARGet) | TARGet Analysis Tool | Fails dependency and test checks |
+| [arjunrajlaboratory/memSeqASEanalysis](https://github.com/arjunrajlaboratory/memSeqASEanalysis) | memSeq ASE Analysis | Fails dependency and test checks |
+
+## Configuration
+
+### Synapse Configuration
+
+1. **Authentication Token**
+   - Set as Nextflow secret:
+   ```groovy
+   // nextflow.config
+   secrets {
+       synapse_token = 'your_token_here'
+   }
+   ```
+   - Or via command line:
+   ```bash
+   nextflow run main.nf --synapse_token 'your_token_here'
```

Comment:
You definitely shouldn't use a Nextflow param to pass authentication tokens in for security reasons. See Nextflow secrets documentation https://www.nextflow.io/docs/latest/secrets.html